src/App.tsx

```diff
@@ -1,190 +1,125 @@
+			{ level: "info", ts: 1547306774.967864, caller: "AnalyseTrainingPharse/analyse_training_pharse.go:74", msg: "Match", question: { qid: "5c24d5f0d251974fc4534876", index: 1, trainingPhrase: "ขอเลขที่ใบเสร็จอีกครั้ง", matched: false, intent: "5c316af134d1f90001aa26df", retry: 0 } },
+			{ level: "info", ts: 1547306775.27901, caller: "AnalyseTrainingPharse/analyse_training_pharse.go:74", msg: "Match", question: { qid: "5c24d5f0d251974fc4534876", index: 2, trainingPhrase: "ขอเลขใบเสร็จใหม่", matched: false, intent: "5c316af134d1f90001aa26df", retry: 0 } },
+			{ level: "info", ts: 1547306775.581335, caller: "AnalyseTrainingPharse/analyse_training_pharse.go:74", msg: "Match", question: { qid: "5c24d5f0d251974fc4534876", index: 3, trainingPhrase: "ขอเลขใบเสร็จอีกครั้ง", matched: false, intent: "5c316af134d1f90001aa26df", retry: 0 } },
+			{ level: "info", ts: 1547306775.888649, caller: "AnalyseTrainingPharse/analyse_training_pharse.go:74", msg: "Match", question: { qid: "5c24d5f0d251974fc4534876", index: 4, trainingPhrase: "อยากได้เลขใบเสร็จใหม่", matched: false, intent: "5c316af134d1f90001aa26df", retry: 0 } },
+			{ level: "info", ts: 1547306776.1961741, caller: "AnalyseTrainingPharse/analyse_training_pharse.go:74", msg: "Match", question: { qid: "5c24d5f0d251974fc4534876", index: 5, trainingPhrase: "เอาเลขใบเสร็จอีกรอบ", matched: false, intent: "5c316af134d1f90001aa26df", retry: 0 } },
+			{ level: "info", ts: 1547306776.4603899, caller: "AnalyseTrainingPharse/analyse_training_pharse.go:74", msg: "Match", question: { qid: "5c24d5f0d251974fc4534876", index: 6, trainingPhrase: "ขอเลขใบเสร็จอีกรอบ", matched: false, intent: "5c316af134d1f90001aa26df", retry: 1 } },
+			{ level: "info", ts: 1547306776.708064, caller: "AnalyseTrainingPharse/analyse_training_pharse.go:74", msg: "Match", question: { qid: "5c24d5f0d251974fc4534876", index: 7, trainingPhrase: "วิธี running ใบเสร็จใหม่", matched: false, intent: "5c316af134d1f90001aa26df", retry: 0 } },
+			{ level: "info", ts: 1547306777.023008, caller: "AnalyseTrainingPharse/analyse_training_pharse.go:74", msg: "Match", question: { qid: "5c24d5f0d251974fc4534876", index: 8, trainingPhrase: "รันเลขที่ใบเสร็จเลขใหม่ให้หน่อยค่ะ เนื่องจากใบเสร็จรับเงินล็อตเก่าหมด", matched: false, intent: "5c316af134d1f90001aa26df", retry: 0 } },
+		],
+		name: "default.json",
+	};
+
+	private defaultCols: DataTypes.SelectOption[] = [
+		{ value: "level", label: "level" },
+		{ value: "ts", label: "ts" },
+		{ value: "caller", label: "caller" },
+		{ value: "msg", label: "msg" },
+		{ value: "question", label: "question" },
+		{ value: "count", label: "count" },
+		{ value: "test", label: "test" },
+		{ value: "test2", label: "test2" },
+		{ value: "test3", label: "test3" },
+	];
+
+	public constructor(props: {}) {
```

Comment:
`public constructor`?

Reply:
Is it difference? Since it's only effect compile and not effect runtime? 
Should it be `public` or `private`??

Reply:
or not both?